enhancements/sig-policy/122-expanded-hub-templates-access/README.md

```diff
@@ -0,0 +1,310 @@
+# Expand Hub Template Access on Policies
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in
+      [website](https://github.com/open-cluster-management-io/open-cluster-management-io.github.io/)
+
+## Summary
+
+Open Cluster Management policy hub templates are limited to referencing objects in the same namespace as the root
+policy. The proposal is to add an additional field of `spec.hubTemplateOptions.serviceAccountName` that must reference a
+service account in the same namespace as the root policy. The hub templates would execute using that service account.
+The user is responsible for providing `list` and `watch` permissions on the objects that the hub templates reference for
+that service account.
+
+## Motivation
+
+An Open Cluster Management policy can contain hub templates that get executed as replicated policies are created in each
+managed cluster namespace on the hub. To prevent privilege escalation and data leakage, there is a current restriction
+that only the `ManagedCluster` object and objects in the same namespace as the root policy can be referenced. This is
+limiting in situations such as copying a `Secret` from another namespace to the managed cluster. The current workaround
+is to have a policy applied to `local-cluster` (Hub) that copies the `Secret` to the root policy namespace. Workarounds
+for cluster-scoped objects or objects that have side effects are more complicated.
+
+### Goals
+
+1. A user can specify a service account for hub templates to access objects outside of the root policy namespace.
+1. The current event-driven nature of hub templates is retained with a custom service account.
+1. Tokens for the service account have an expiration of 2 hours.
+
+### Non-Goals
+
+1. Allowing service accounts outside of the root policy namespace.
+
+## Proposal
+
+### Design
+
+Define a new field at `spec.hubTemplateOptions.serviceAccountName` to specify a service account that exists in the root
+policy namespace. When specified, the service account is used instead of the service account of the Governance Policy
+Propagator for hub template execution. Additionally, no namespace or cluster-scoped restrictions will be enforced by the
+Governance Policy Propagator, and instead, it will rely on the Kubernetes API performing authorization checks of the
+service account. Note that due to the event-driven nature of hub templates, the service account must have `list` and
+`watch` permissions for the objects that the hub templates reference.
+
+Note that for backwards compatibility, if the namespace is not specified in a hub template object lookup, the root
+policy namespace will be the default, however, the service account must still have the appropriate permissions in the
```

Comment:
So in the example, the following function:
```
                  field3: '{{hub fromConfigMap "" "test-configmap" "test-field1" hub}}'
```
would default to the test-policies namespace and would require list and watch permissions for ConfigMaps in the test-policies namespace to be in the roles for the serviceaccount.

Reply:
Yes, that's right.